src/CQRS/LeanCode.CQRS.AspNetCore/Registration/CQRSObjectsRegistrationSource.cs

```diff
@@ -18,16 +18,12 @@ public CQRSObjectsRegistrationSource(IServiceCollection services)
         this.services = services;
     }
 
-    public void AddCQRSObjects(TypesCatalog contractsCatalog, TypesCatalog handlersCatalog)
+    public void AddCQRSObjects(IEnumerable<TypeInfo> possibleContracts, IEnumerable<TypeInfo> possibleHandlers)
```

Comment:
I'm against this change - if we want to allow registering limited number of types, then the `TypesCatalog` should change, not every application of it.

Reply:
My take here is given the limited applicability of manual registration, it's better to have explicit
`AddCommand<TCommand,THandler>()` etc. methods.

By the way, the goal of this change was to facilitate open generic handlers (with user id type specified by the application), doing it via `TypesCatalog` would be awkward, as you'd need to find types you'd need to close (thus explicit manual registration).

Reply:
You can't have open generics here - you won't have enough data to close them. Handlers must, in the end, be closed from the get go to be instantiatable by simple `GetService<ICommandHandler<XYZ>>()` calls - the call doesn't have the necessary types specified to close anything.

In this particular case, you _know_ what type the handler must close on, because it is constant for the project, i.e. the handler is `RegisterTokenCH<TUserId>`, but you will register only `RegisterTokenCH<Guid>` or `RegisterTokenCH<string>`.

You will basically end up with `new TypesCatalog().Add<RegisterTokenCH<Guid>>()`. It won't be that different from `AddCommand` that you propose, and TBH I'm indifferent here. `AddCommand`/`AddQuery` is also OK for me. I just don't want to have an option to add a set of queries/commands by passing list.